aiocogeo/partial_reads.py

```diff
@@ -119,18 +119,9 @@ def _get_overview_level(
         )
         target_res = target_gt.a
 
-        ovr_level = 0
-        if target_res > src_res:
-            # Decimated resolution at each overview
-            overviews = [src_res * decim for decim in self.overviews]
-            for ovr_level in range(ovr_level, len(overviews) - 1):
-                ovr_res = src_res if ovr_level == 0 else overviews[ovr_level]
-                if (ovr_res < target_res) and (overviews[ovr_level + 1] > target_res):
-                    break
-                if abs(ovr_res - target_res) < 1e-1:
-                    break
-            else:
-                ovr_level = len(overviews) - 1
+        # Choose overview whose res is closest to the target res. This can lead to downsampling or upsampling.
+        available_resolutions = np.array([src_res] + [src_res * decim for decim in self.overviews])
+        ovr_level = np.argmin(np.abs(available_resolutions - target_res))
```

Comment:
This will choose the closest available resolution, regardless of whether that is higher or lower than the target resolution.

Is it acceptable to upsample in this way? An alternative is to pick the largest available resolution where `available_res < target_res + lambda` so we would only downsample. But this could 4x the amount of data for many requests with barely perceptible impact on quality. We could consider adding this as an option/environment variable?

Reply:
Sorry I'm slow to get to this PR!

> We could consider adding this as an option/environment variable?

I think this is the best option, there are use cases where it isn't acceptable to upsample at all so it makes sense for the user to be able to choose.  This is also a bigger problem at higher zooms (where the absolute difference in resolution between zooms becomes quite low); at higher zooms you're more likely to fall back to a lower zoom that is less efficient from a tiling perspective.

I'm not sure what the "sensible default" for such an environment variable would be (allow upsampling or not).  I think there are good arguments for either.  Not allowing upsampling is probably the most intuitive but as you mention leads to not so intuitive performance losses.  I lean towards favoring performance, as the primary goal of the library is to provide higher performance over other libs (GDAL etc.).